File: LeetCode/125. Valid Palindrome.js
// https://leetcode.com/problems/valid-palindrome/description/

const isPalindrome = (str) => {
  const serialized = str.replaceAll(/\W|_/gi, '').toLowerCase();

  return serialized === serialized.split('').reverse().join('');
};

console.log(isPalindrome('A man, a plan, a canal: Panama')); // true
console.log(isPalindrome('race a car')); // false
console.log(isPalindrome(' ')); // true

File: LeetCode/125. Valid Palindrome.py
# https://leetcode.com/problems/valid-palindrome/description/


class Solution:
    def isPalindrome(self, input: str) -> int:
        serialized = "".join(
            [value for value in input if value.isalpha() or value.isdigit()]
        ).lower()

        return serialized == serialized[::-1]
